packages/compass-preferences-model/src/preferences-schema.ts

```diff
@@ -1027,6 +1082,21 @@ function deriveNetworkTrafficOptionState<K extends keyof AllPreferences>(
   });
 }
 
+/** Helper for deriving value/state for preferences from other preferences */
+function deriveValueFromOtherPreferencesAsLogicalAnd<
```

Comment:
Can we add a unit test or two for this function? Logic looks good but I think it's good to test something like this.

Reply:
Ah just noticed it's covered already in preferences.spec.ts 👍  Thanks for the callout